docs/faas/aws-lambda.md

```diff
@@ -54,21 +54,21 @@ and the [cloud resource conventions][cloud]. The following AWS Lambda-specific a
 [faasres]: /docs/resource/faas.md (FaaS resource conventions)
 [cloud]: /docs/resource/cloud.md (Cloud resource conventions)
 
-### AWS X-Ray Environment Span Link
+### AWS X-Ray Environment `SpanContext`
 
-If the `_X_AMZN_TRACE_ID` environment variable is set, instrumentation SHOULD try to parse an
-OpenTelemetry `Context` out of it using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.22.0/specification/context/api-propagators.md). If the
-resulting `Context` is [valid](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.22.0/specification/trace/api.md#isvalid) then a [Span Link][] SHOULD be added to the new Span's
-[start options](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.22.0/specification/trace/api.md#specifying-links) with an associated attribute of `source=x-ray-env` to
-indicate the source of the linked span.
-Instrumentation MUST check if the context is valid before using it because the `_X_AMZN_TRACE_ID` environment variable can
-contain an incomplete trace context which indicates X-Ray isn’t enabled. The environment variable will be set and the
-`Context` will be valid and sampled only if AWS X-Ray has been enabled for the Lambda function. A user can
-disable AWS X-Ray for the function if the X-Ray Span Link is not desired.
+If the `_X_AMZN_TRACE_ID` environment variable is set, instrumentation MUST ensure that the value from that
+environment variable is included in the `Carrier` used to extract a parent `Context` with the `Field` name
+"X-Amzn-Trace-Id" unless the `Carrier` already includes a value for that `Field`.
 
-**Note**: When instrumenting a Java AWS Lambda, instrumentation SHOULD first try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) before checking and attempting to parse the environment variable above.
+Instrumentation MUST provide users with the option to create a `Link` to the AWS Lambda vended spans.  If this
```

Comment:
This may actually need to be a general convention for the Propagators framework.

Effectively, I think this should be the defacto behavior when multiple propagators of *trace* exist and they all find parent spans.

Imagine the user configures:

```
propagators:
  - w3c
  - baggage
  - xray
```

This would mean:

- When just x-ray *or* w3c propagation exist, you would attach-to-parent against x-ray or w3c normally.
- When *both* x-ray and w3c propagation exist, you would attach-to-parent w/ w3c and link-to x-ray.

This would help across more than just AWS, as you could imagine a x-cloud application leverage the GCP format could do the following:

```
propagators:
  - w3c
  - baggage
  - xray
  - gcp
```

Which, effectively, would give users the best possible way to deal with the reality of security-boundaries and broken trace parenting.

Reply:
> This may actually need to be a general convention for the Propagators framework.
> 
> Effectively, I think this should be the defacto behavior when multiple propagators of _trace_ exist and they all find parent spans.

I don't disagree, though I expect doing so would require significant changes to the [composite propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#composite-propagator) specification.  At present that specification provides no mechanism for determining which of the composed propagators produced the resulting context and has no way to identify the individual propagators is composes.  I'd like to not have this change held up by needing to make what would undoubtedly be breaking changes to stable APIs.